api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -314,59 +314,37 @@ private RenewableAccessModuleStatus mkStatus(ModuleNameEnum name, ModuleTimes ti
   }
 
   private boolean isDataUseAgreementCompliant(DbUser user) {
-    final ModuleTimes duaTimes =
-        new ModuleTimes(
-            user.getDataUseAgreementCompletionTime(), user.getDataUseAgreementBypassTime());
-    if (duaTimes.isBypassed()) {
+    final Integer signedVersion = user.getDataUseAgreementSignedVersion();
+    if (accessModuleService.isModuleBypassed(user, AccessModuleName.DATA_USER_CODE_OF_CONDUCT)) {
       return true;
     }
-    final Integer signedVersion = user.getDataUseAgreementSignedVersion();
     if (signedVersion == null || signedVersion != getCurrentDuccVersion()) {
       return false;
     }
-    return duaTimes.isComplete() && !duaTimes.hasExpired();
-  }
-
-  // Checking for annual completion time is not a part of this module
-  private boolean isEraCommonsCompliant(DbUser user) {
-    return user.getEraCommonsBypassTime() != null
-        || !configProvider.get().access.enableEraCommons
-        || user.getEraCommonsCompletionTime() != null;
-  }
-
-  private boolean isComplianceTrainingCompliant(DbUser user) {
-    if (!configProvider.get().access.enableComplianceTraining) {
-      return true;
-    }
-    final ModuleTimes ctTimes =
-        new ModuleTimes(
-            user.getComplianceTrainingCompletionTime(), user.getComplianceTrainingBypassTime());
-    return ctTimes.isBypassed() || (ctTimes.isComplete() && !ctTimes.hasExpired());
-  }
-
-  private boolean isProfileConfirmationCompliant(DbUser user) {
-    final ModuleTimes profileTimes = new ModuleTimes(user.getProfileLastConfirmedTime(), null);
-    return profileTimes.isComplete() && !profileTimes.hasExpired();
-  }
-
-  private boolean isPublicationConfirmationCompliant(DbUser user) {
-    final ModuleTimes publicationTimes =
-        new ModuleTimes(user.getPublicationsLastConfirmedTime(), null);
-    return publicationTimes.isComplete() && !publicationTimes.hasExpired();
+    return accessModuleService.isModuleCompliant(user, AccessModuleName.DATA_USER_CODE_OF_CONDUCT);
   }
 
   private boolean shouldUserBeRegistered(DbUser user) {
-    // 2FA does not need to be checked for annual renewal
     boolean twoFactorAuthComplete =
-        user.getTwoFactorAuthCompletionTime() != null || user.getTwoFactorAuthBypassTime() != null;
-    // TODO: can take out other checks once we're entirely moved over to the 'module' columns
+        accessModuleService.isModuleCompliant(user, AccessModuleName.TWO_FACTOR_AUTH);
+    boolean eRACommonsComplete =
+        !configProvider.get().access.enableEraCommons
```

Comment:
Right, we do not need enabled check here. I have multiple booleans because it is easy to read the code.
I don't think your simpler stream AllMatch works because:
(1) Sooner we will add CT_TRAINNING module which is for CT
(2) ERA might be optional and I don't think putting inst tier requirement inside isCompliant is good idea
(3) DUCC has version check